internal/monitors/processlist/processlist_windows.go

```diff
@@ -0,0 +1,172 @@
+// https://msdn.microsoft.com/en-us/library/aa394323(v=vs.85).aspx
+type PerfProcProcess struct {
+	IDProcess            uint32
+	PercentProcessorTime uint64
+}
+
+// getCPUPercentages is set as a package variable so we can mock it durring testing
+var getCPUPercentages = func() (cpuPercents map[uint32]uint64, err error) {
+	// Get all process cpu percentages
+	var processes []PerfProcProcess
+	if err = wmi.Query("select IDProcess, PercentProcessorTime from Win32_PerfFormattedData_PerfProc_Process where Name != '_Total'", &processes); err == nil && len(processes) > 0 {
+		cpuPercents = make(map[uint32]uint64, len(processes))
+		for _, p := range processes {
+			cpuPercents[p.IDProcess] = p.PercentProcessorTime
+		}
+	}
+	return cpuPercents, err
+}
+
+// getAllProcesses retrieves all processes.  It is set as a package variable so we can mock it durring testing
+var getAllProcesses = func() (ps []Win32Process, err error) {
+	err = wmi.Query("select Name, ExecutablePath, CommandLine, Priority, ProcessID, Status, ExecutionState, KernelModeTime, PageFileUsage, UserModeTime, WorkingSetSize, VirtualSize from Win32_Process", &ps)
+	return ps, err
+}
+
+// getUsername - retrieves a username from an open process handle it is set as a package variable so we can mock it durring testing
+var getUsername = func(id uint32) (username string, err error) {
+	// open the process handle and collect any information that requires it
+	var h windows.Handle
+	defer windows.CloseHandle(h)
+	if h, err = windows.OpenProcess(processQueryLimitedInformation, false, id); err != nil {
+		err = fmt.Errorf("unable to open process handle. %v", err)
+		return username, err
+	}
+
+	// the windows api docs suggest that windows.TOKEN_READ is a super set of windows.TOKEN_QUERY,
+	// but in practice windows.TOKEN_READ seems to be less permissive for the admin user
+	var token windows.Token
+	defer token.Close()
+	err = windows.OpenProcessToken(h, windows.TOKEN_QUERY, &token)
+	if err != nil {
+		err = fmt.Errorf("unable to retrieve process token. %v", err)
+		return username, err
+	}
+
+	// extract the user from the process token
+	user, err := token.GetTokenUser()
+	if err != nil {
+		err = fmt.Errorf("unable to get token user. %v", err)
+		return username, err
+	}
+
+	// extract the username and domain from the user
+	userid, domain, _, err := user.User.Sid.LookupAccount("")
+	if err != nil {
+		err = fmt.Errorf("unable to look up user account from Sid. %v", err)
+	}
+	username = fmt.Sprintf("%s\\%s", domain, userid)
+
+	return username, err
+}
+
+// ProcessList takes a snapshot of running processes and returns a byte buffer
+func ProcessList() (*bytes.Buffer, error) {
+	processes := &bytes.Buffer{}
+	processes.WriteString("{")
+	defer processes.WriteString("}") // always close the associative array
+
+	// Get all processes
+	ps, err := getAllProcesses()
+	if err != nil {
+		logger.Debugf("no processes returned %v", err)
```

Comment:
This indicates something pretty bad going on right since there should be at least one process running in the system?